hack/local-e2e-common.sh

```diff
@@ -0,0 +1,474 @@
+#!/bin/bash
+# Common functions and setup for local e2e test scripts
+# This script is sourced by local-e2e-test.sh and local-e2e-pool-test.sh
+
+# Supported cloud platforms (keep in sync with Makefile test-e2e-local target)
+SUPPORTED_CLOUDS="aws/azure/gcp/ibmcloud/vsphere/nutanix/openstack"
+
+# Setup and validate KUBECONFIG
+# If KUBECONFIG environment variable is set, use it; otherwise use default ~/.kube/config
+function setup_kubeconfig() {
+    if [ -n "$KUBECONFIG" ]; then
+        echo "Using KUBECONFIG from environment: $KUBECONFIG"
+        if [ ! -f "$KUBECONFIG" ]; then
+            echo "Error: KUBECONFIG file not found: $KUBECONFIG" >&2
+            exit 1
+        fi
+    else
+        # Auto-detect: use default kubeconfig location if KUBECONFIG is not set
+        DEFAULT_KUBECONFIG="${HOME}/.kube/config"
```

Comment:
I appreciate that we're trying to make this as seamless as possible, but since this test is not guaranteed to be nondestructive, I don't think we should try so hard to discover a default.

In fact, I'm not super convinced we should even use whatever `$KUBECONFIG` is currently set in the env -- we should consider requiring it to be provided explicitly, either via a specially-named env var (`$HIVE_E2E_HUB_KUBECONFIG`?) or a CLI argument (`--hub-kubeconfig`). If they want to use their `$KUBECONFIG` they can always do so (`--hub-kubeconfig $KUBECONFIG`).